components/core/src/clp_s/TraceableException.hpp

```diff
@@ -8,42 +8,43 @@
 
 #include "ErrorCode.hpp"
 
-#define __FILENAME__ ((__FILE__) + SOURCE_PATH_SIZE)
+// Define a version of __FILE__ that's relative to the source directory
+// NOLINTBEGIN(bugprone-reserved-identifier,cert-dcl37-c,cert-dcl51-cpp,readability-identifier-naming)
```

Comment:
Since this is only necessary for the `__FILENAME__` line, I'd prefer we change it to a `NOLINTNEXTLINE` and move it just above the `__FILENAME__` line so we can reduce the scope of the no-lint section.